lib/src/configuration.rs

```diff
@@ -0,0 +1,53 @@
+#[derive(Debug, Clone)]
+pub struct Configuration {
+    pub base_path: String,
+    pub user_agent: Option<String>,
+    pub basic_auth: Option<BasicAuth>,
+    pub oauth_access_token: Option<String>,
+    pub bearer_access_token: Option<String>,
+    pub api_key: Option<ApiKey>,
+    // TODO: take an oauth2 token source, similar to the go one
+}
+
+pub type BasicAuth = (String, Option<String>);
+
+#[derive(Debug, Clone)]
+pub struct ApiKey {
+    pub prefix: Option<String>,
+    pub key: String,
+}
+
+impl Configuration {
+    pub fn new(
+        base_path: String,
+        user_agent: Option<String>,
+        oauth_access_token: Option<String>,
+    ) -> Self {
+        Configuration {
+            base_path,
+            user_agent,
+            basic_auth: None,
+            oauth_access_token,
+            bearer_access_token: None,
+            api_key: None,
+        }
+    }
+
+    pub fn set_base_path(&mut self, base_path: &str) -> &mut Self {
+        self.base_path = base_path.to_string();
+        self
+    }
+}
+
+impl Default for Configuration {
```

Comment:
I don't really understand this. Seems to me that we have a constructor that requires 3 args, and a default configuration. But what if I want to create an instance and provide only some args. Can't I set defaults for every value that wasn't provided as an arg instead?